jquery.flot.js

```diff
@@ -1030,17 +1036,30 @@ Licensed under the MIT license.
             });
 
             // The outermost axis on each side has no margin
-
             if (outermost) {
                 axisMargin = 0;
             }
 
             // Set the default tickLength if necessary
-
             if (tickLength == null) {
                 tickLength = 10;
             }
 
+            // By default, major tick marks are visible
+            if(showTicks == null) {
+                showTicks = true;
+            }
+
+            // By default, minor tick marks are visible
+            if(showMinorTicks == null) {
+                showMinorTicks = true;
+            }
+
+            // By default, grid lines are visible
+            if (gridLines == null) {
+                gridLines = true;
+            }
+
             if (!isNaN(+tickLength))
                 padding += +tickLength;
```

Comment:
Not directly related to your changes, but are these lines working fine? Those 2 double tickLength increments look suspicious.